lib/puppet/provider/ssh_authorized_key/parsed.rb

```diff
@@ -38,7 +38,11 @@ def dir_perm
     0o700
   end
 
-  def file_perm
+  def file_perm_readonly
+    0o444
```

Comment:
I know multiple organisations where world readable files aren't allowed. I don't think it's a good idea to implement such a breaking change. I think it makes more sense to add a parameter for the mode and default to `0600` for now.

Reply:
hmm... sure, but how would that even work considering multiple invocations of the type can have different mode parameters? how would we handle that conflict? we'd just let puppet flap?

Reply:
In that case puppet flaps or even better: the type notices it and fails.